### PR TITLE
Add missing woocommerce_run_on_woocommerce_admin_updated hook for RemoteInboxNotificationsEngine scheduled action

### LINKED ISSUE #36767: woocommerce_run_on_woocommerce_admin_updated hook doesn't run on update
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

woocommerce_run_on_woocommerce_admin_updated doesn't run on an update with the following error message

`action failed via WP Cron: Scheduled action for woocommerce_run_on_woocommerce_admin_updated will not be executed as no callbacks are registered.`

### Expected behavior

woocommerce_run_on_woocommerce_admin_updated should run on an update

### Actual behavior

woocommerce_run_on_woocommerce_admin_updated doesn't run on an update

### Steps to reproduce

1. Go to `WooCommerce -> Status -> Scheduled Actions` and search for `woocommerce_run_on_woocommerce_admin_updated` then delete it
2. Open `wp_options` table and search for `woocommerce_version`
3. Update `woocommerce_version` to a lower number.
4. Go to any WooCommerce Admin page and refresh.
5. Confirm `woocommerce_version` has been updated to the latest number.
6. Go to `WooCommerce -> Status -> Scheduled Actions` and search for `woocommerce_run_on_woocommerce_admin_updated`. 
7. Confirm the error message.

### WordPress Environment

latest

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

-   [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds missing `woocommerce_run_on_woocommerce_admin_updated` hook to fix scheduled action error.

Closes #36767 .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Go to `WooCommerce -> Status -> Scheduled Actions` and search for `woocommerce_run_on_woocommerce_admin_updated` then delete it
2. Open `wp_options` table and search for `woocommerce_version`
3. Update `woocommerce_version` to a lower number.
4. Go to any WooCommerce Admin page and refresh.
5. Confirm `woocommerce_version` has been updated to the latest number.
6. Go to `WooCommerce -> Status -> Scheduled Actions` and search for `woocommerce_run_on_woocommerce_admin_updated`. 
7. Confirm the action executed.

### Other information:

-   [X] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php [hunk 1]
```diff
@@ -40,19 +40,20 @@ public static function init() {
...
 				);
 				if ( $next_hook === null ) {
 					WC()->queue()->schedule_single(
 						time(),
 						'woocommerce_run_on_woocommerce_admin_updated',
-						array( __CLASS__, 'run_on_woocommerce_admin_updated' ),
+						array(),
 						'woocommerce-remote-inbox-engine'
 					);
 				}
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php]: 🤔 Assuming the remote inbox engine is only initialized once, we might possibly be able to simplify this whole block of code to something like:

```
WC()->queue()->schedule_single(
 	time(),
 	'woocommerce_updated',
 	array( __CLASS__, 'run_on_woocommerce_admin_updated' ),
 	array(),
 	'woocommerce-remote-inbox-engine'
 );
 ```
 
 I think that shows the intent of the code more clearly and avoids needing the extra action (`woocommerce_run_on_woocommerce_admin_updated`).

Comment (moon0326) [reply plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php]: That's a good question!

Updated in https://github.com/woocommerce/woocommerce/pull/36768/commits/580b9ba069ccabe3936177fbf221b9d107c913aa

**Comment by moon0326** [comment]
@woocommerce/mothra ✋ Could I get a review on this please?

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php]: Hey @adrianduffell ✋ 

Could you take a look at it one more time please?

**Review by adrianduffell — APPROVED** [review]
This tested well and LGTM! 🚀 

Thanks for your work on this and sorry for the super late re-review

Let's merge with one review due to how simple it is.

Could you also pls double-check the testing instructions on the final step? I think `woocommerce_updated` is now the term to search for.

**Comment by adrianduffell** [comment]
Merging on behalf of @moon0326

**Pull request merged**